tapir/shifts/models.py

```diff
@@ -1208,3 +1208,66 @@ class SolidarityShift(models.Model):
     is_used_up = models.BooleanField(default=False)
     date_gifted = models.DateField(auto_now_add=True)
     date_used = models.DateField(null=True)
+
+
+class StaffingEventsChoices(models.TextChoices):
+    ALMOST_FULL = "ALMOST_FULL", _("Shift is almost full, only one spot left.")
+    FULL = "FULL", _("Shift is full now.")
+    UNDERSTAFFED = "UNDERSTAFFED", _("The Shift is understaffed!")
+    ALL_CLEAR = "ALL_CLEAR", _(
+        "All clear: The shift is no longer understaffed, but it's not fully staffed yet either..."
+    )
+    ATTENDANCE_PLUS = "SLOTS_PLUS", _(
+        "One new attendance or more registered, but the shift is neither understaffed nor full or almost full."
+    )
+    ATTENDANCE_MINUS = "SLOTS_MINUS", _(
+        "One attendance or more un-registered, but the shift is neither understaffed nor full or almost full."
+    )
+    SHIFT_COORDINATOR_MINUS = "SHIFT_COORDINATOR_MINUS", _(
+        "The Shift Coordinator has unregistered"
+    )
+    SHIFT_COORDINATOR_PLUS = "SHIFT_COORDINATOR_PLUS", _(
+        "A Shift Coordinator has registered"
+    )
+
+
+def get_staffingevent_choices():
+    return StaffingEventsChoices
+
+
+def get_staffingevent_defaults():
+    return [StaffingEventsChoices.FULL, StaffingEventsChoices.UNDERSTAFFED]
+
+
+class ShiftWatch(models.Model):
+    user = models.ForeignKey(
+        TapirUser, related_name="user_watching_shift", on_delete=models.CASCADE
+    )
+    shift = models.ForeignKey(Shift, on_delete=models.CASCADE)
+    staffing_events = ArrayField(
+        models.CharField(max_length=30, choices=get_staffingevent_choices),
```

Comment:
Could this just be `choices=StaffingEventsChoices.choices`? I'm not sure what `get_staffingevent_choices` is doing.